File: loopr-ios/Models/Trend.swift
//
//  Trend.swift
//  loopr-ios
//
//  Created by kenshin on 2018/3/19.
//  Copyright © 2018年 Loopring. All rights reserved.
//

import Foundation

class Trend {
    let amount: Double
    let close: Double
    let market: String
    let start: UInt
    let end: UInt
    let intervals: String
    let open: Double
    var low: Double
    let createTime: UInt
    var high: Double
    let vol: Double

	init(json: JSON) {
		self.amount = json["amount"].doubleValue
		self.close = json["close"].doubleValue
		self.market = json["market"].stringValue
		self.intervals = json["intervals"].stringValue
		self.open = json["open"].doubleValue
		self.low = json["low"].doubleValue
        self.high = json["high"].doubleValue
        self.vol = json["vol"].doubleValue
        self.start = json["start"].uIntValue
        self.end = json["end"].uIntValue
        self.createTime = json["createTime"].uIntValue
        
        // Have to tune the data to avoid abnormal market orders.
        if high > 1.25 * max(open, close) {
            high = 1.25 * max(open, close)
        }
        
        if low < 0.8 * min(open, close) {
            low = 0.8 * min(open, close)
        }
	}
}

File: walletsdk/src/main/java/leaf/prod/walletsdk/model/Trend.java
/**
 * Created with IntelliJ IDEA.
 * User: kenshin wangchen@loopring.org
 * Time: 2019-01-12 2:42 PM
 * Cooperation: loopring.org 路印协议基金会
 */
package leaf.prod.walletsdk.model;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class Trend {

    private String intervals;

    private String market;

    private Double vol;

    private Double amount;

    private Double createTime;

    private Double open;

    private Double close;

    private Double high;

    private Double low;

    private Long start;

    private Long end;

    private String change;

    private String range;
}
